### ISSUE TITLE
[enh] Audio Graph

### ISSUE THREAD

**Issue by Larpoux** [Canardoux/flutter_sound#479]
Actually, looping from Recorder to Player inside Dart code is not fine : 
the playback is always too late compared to the recording.
And the delay is more and more unacceptable.

We need something like a direct link between the Recorder and the Player inside the OS, without involving Flutter.

But probably we will have the same problem when the App tries to record to a stream, or playback from a stream.
The communication between Dart and the OS does not seem efficient enough.
This is probably a **BIG** problem.
[see #90](https://github.com/dooboolab/flutter_sound/issues/90)

**Comment by RevanthRameshkumar** [comment]
@Larpoux, this person made a guitar tuner using platform channels. https://pub.dev/packages/flutter_fft
Is this implementation different from yours? Would this implementation style help w/ the delay issue?

**Comment by Larpoux** [comment]
Please refer to [#90]. I just posted something.

**Comment by Larpoux** [comment]
Yes, FFI is probably necessary for this. But I really think that most of the work must be done inside the &tau; Core, and not the flutter side.

**Comment by Larpoux** [comment]
Actually I am working on a latency problem and I hope to be able to improve it.
I plan to work very soon on implementing the concept of audio-graph.

Audio-graph will be several audio-nodes linked together.
For example, you will be able to connect the Microphone node to the Equalizer node, and the Equalizer node to the Headset node.

I really want to do that. This project is really interesting to be developed, and I am fed up to work on uninteresting things.

My dream is also to create a graphic-editor to be be able to draw the audio-graph with the mouse.
The graphic will generate both a Dart (or Javascript) module and a visual SVG image that the user will be able to insert in its own documentation.
Probably not very useful for &tau; users, but funny for me to develop

**Comment by Larpoux** [comment]
I added an item in [this post](https://tau.canardoux.xyz/commento-post.html)
